### PR TITLE
[docs] Using `expo-gl` inside Reanimated worklets

### PR DESCRIPTION
**Pull request by wkozyra95**
# Why

How to use expo-gl inside reanimated worklets

# How

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/gl-view.md [hunk 1]
```diff
@@ -187,6 +187,49 @@ The `pixels` argument of [`texImage2D()`](https://developer.mozilla.org/en-US/do
 
 For efficiency reasons the current implementations of the methods don't perform type or bounds checking on their arguments. So, passing invalid arguments could cause a native crash. We plan to update the API to perform argument checking in upcoming SDK versions. Currently the priority for error checking is low since engines generally don't rely on the OpenGL API to perform argument checking and, even otherwise, checks performed by the underlying OpenGL ES implementation are often sufficient.
 
+## Integration with reanimated worklets
```
Comment (Simek) [inline docs/pages/versions/unversioned/sdk/gl-view.md:190]:
```suggestion
## Integration with Reanimated worklets
```
Just a small suggestion. 😄

File: docs/pages/versions/unversioned/sdk/gl-view.md [hunk 2]
```diff
@@ -187,6 +187,49 @@ The `pixels` argument of [`texImage2D()`](https://developer.mozilla.org/en-US/do
...
+
+To use this API inside Reanimated worklet you need to pass context id to the worklet code and restore context like in the example bellow.
+
+```js
```
Comment (Simek) [inline docs/pages/versions/unversioned/sdk/gl-view.md:194]: Can we convert this example to `InlineSnack`?

**Review by Simek — APPROVED** [review]
I have left one, small suggestion and a question, besides that LGTM!

File: docs/pages/versions/unversioned/sdk/gl-view.md [hunk 3]
```diff
@@ -187,6 +187,54 @@ The `pixels` argument of [`texImage2D()`](https://developer.mozilla.org/en-US/do
 
 For efficiency reasons the current implementations of the methods don't perform type or bounds checking on their arguments. So, passing invalid arguments could cause a native crash. We plan to update the API to perform argument checking in upcoming SDK versions. Currently the priority for error checking is low since engines generally don't rely on the OpenGL API to perform argument checking and, even otherwise, checks performed by the underlying OpenGL ES implementation are often sufficient.
 
+## Integration with Reanimated worklets
+
+To use this API inside Reanimated worklet you need to pass context id to the worklet code and restore context like in the example bellow.
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/gl-view.md:192]:
```suggestion
To use this API inside Reanimated worklet you need to pass the GL context ID to the worklet and recreate the GL object like in the example bellow.
```

File: docs/pages/versions/unversioned/sdk/gl-view.md [hunk 4]
```diff
@@ -187,6 +187,54 @@ The `pixels` argument of [`texImage2D()`](https://developer.mozilla.org/en-US/do
...
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+      <GLView style={{ width: 300, height: 300 }} onContextCreate={onContextCreate} />
+    </View>
+  );
+}
+```
+
+</SnackInline>
+
+For more in depth example on how to use `expo-gl` with Reanimated and Gesture Handler you can check [here](https://github.com/expo/expo/blob/master/apps/native-component-list/src/screens/GL/GLReanimatedExample.tsx).
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/gl-view.md:226]:
```suggestion
For more in-depth example on how to use `expo-gl` with Reanimated and Gesture Handler you can check [this example](https://github.com/expo/expo/blob/master/apps/native-component-list/src/screens/GL/GLReanimatedExample.tsx).
```

File: docs/pages/versions/unversioned/sdk/gl-view.md [hunk 5]
```diff
@@ -187,6 +187,54 @@ The `pixels` argument of [`texImage2D()`](https://developer.mozilla.org/en-US/do
...
+
+### Limitations
+
+Worklet runtime is imposing some limitations on the code that runs inside it, so if you have existing WebGL code, it'll likely require some modifications to run inside a worklet thread.
+- Third-party libraries like Pixi.js or Three.js won't work inside the worklet, you can only use functions that have `'worklet'` added at the start.
+- If you need to load some assets to pass to the WebGL code, it needs to be done on the main thread and passed via some reference to the worklet. If you are using `expo-assets` you can just pass asset object returned by `Asset.fromModule` or from hook `useAssets` to the `runOnUI` function.
+- To implement a rendering loop you need to use `requestAnimationFrame`, APIs like `setTimeout` are not supported.
+- It's supported only on Android and iOS, it won't work in browser.
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/gl-view.md:234]:
```suggestion
- It's supported only on Android and iOS, it doesn't work on Web.
```

**Review by tsapeta — APPROVED** [review]

**Pull request merged**